auto_submit/lib/validations/revert.dart

```diff
@@ -100,4 +129,73 @@ class Revert extends Validation {
     List<String> linkSplit = link.split('#');
     return int.parse(linkSplit.elementAt(1));
   }
+
+  /// Wait for the required checks to complete.
+  Future<bool> waitForRequiredChecks({
+    required GithubService githubService,
+    required github.RepositorySlug slug,
+    required String sha,
+    required List<String> checkNames,
+  }) async {
+    List<github.CheckRun> targetCheckRuns = [];
+    for (var element in checkNames) {
+      targetCheckRuns.addAll(
+        await githubService.getCheckRunsFiltered(
+          slug: slug,
+          ref: sha,
+          checkName: element,
+        ),
+      );
+    }
+
+    bool checksCompleted = true;
+
+    try {
+      for (github.CheckRun checkRun in targetCheckRuns) {
+        await _waitForRequiredCheckRunsWithRetries(
+          () async {
+            await _verifyCheckRunCompleted(
+              slug,
+              githubService,
+              checkRun,
+            );
+          },
+          retryOptions,
+        );
+      }
+    } catch (e) {
+      log.warning('Required check has not completed in time. ${e.toString()}');
```

Comment:
What happens if we enable autosubmit on non-flutter repos that would not have the ci.yaml validation check?

Reply:
That is a good question. Currently this will be an issue. We need to map the required tests to repositories since github does not expose required checks.

Reply:
After changing to a mapping with repository name, this will work fine. If a repository does not have any required checks it would not run the retry function. The drawback to this is that since github does not expose required checks we have to keep track of this manually within the source.